code/frameworks/svelte-vite/src/plugins/svelte-docgen.ts

```diff
@@ -44,10 +61,26 @@ function getNameFromFilename(filename: string) {
 
 export function svelteDocgen(svelteOptions: Record<string, any> = {}): PluginOption {
   const cwd = process.cwd();
-  const { preprocess: preprocessOptions, logDocgen = false } = svelteOptions;
+  const { preprocess: preprocessOptions, docPreprocess, logDocgen = false } = svelteOptions;
   const include = /\.(svelte)$/;
   const filter = createFilter(include);
 
+  let docPreprocessOptions = docPreprocess;
+  if (!docPreprocessOptions && preprocessOptions) {
+    /*
+     * We can't use vitePreprocess() for the documentation.
+     * This preprocessor uses esbuild which removes jsdoc.
```

Comment:
clarify which one "this preprocessor" refers to
```suggestion
     * We can't use vitePreprocess() for the documentation
     * because it uses esbuild which removes jsdoc.
```